vimeo-networking/src/main/java/com/vimeo/networking/model/Comment.java

```diff
@@ -75,12 +90,126 @@ public boolean equals(Object o) {
+            }
+
+            out.endObject();
+        }
+
+        @Override
+        public Comment read(JsonReader in) throws IOException {
+            final Comment comment = new Comment();
+            in.beginObject();
+            while (in.hasNext()) {
+                String nextName = in.nextName();
+                JsonToken jsonToken = in.peek();
+                if (jsonToken == JsonToken.NULL) {
+                    in.skipValue();
+                    continue;
+                }
+                switch (nextName) {
+                    case "uri":
+                        comment.uri = in.nextString();
+                        break;
+                    case "text":
+                        comment.text = in.nextString();
+                        break;
+                    case "type":
+                        comment.type = sCommentTypeEnumTypeAdapter.read(in);
+                        break;
+                    case "created_on":
+                        try {
+                            comment.createdOn = ISO8601.toDate(in.nextString());
+                        } catch (ParseException e) {
+                            ClientLogger.e("Error parsing Comment date", e);
+                        }
+                        break;
+                    case "user":
+                        comment.user = mUserTypeAdapter.read(in);
+                        break;
+                    case "metadata":
+                        comment.metadata = mMetadataTypeAdapter.read(in);
+                        break;
+                    default:
+                        // skip any values that we do not account for, without this, the app will crash if the
+                        // api provides more values than we have! [KZ] 6/15/16
+                        in.skipValue();
+                        break;
+                }
+            }
+            in.endObject();
+            return comment;
+        }
+    }
+
+    public static class Factory implements TypeAdapterFactory {
+
+        @SuppressWarnings("unchecked")
+        @Override
+        public <T> TypeAdapter<T> create(Gson gson, TypeToken<T> typeToken) {
+            if (Comment.class.isAssignableFrom(typeToken.getRawType())) {
+                TypeAdapter userTypeAdapter = gson.getAdapter(User.class);
+                TypeAdapter metadataTypeAdapter = gson.getAdapter(Metadata.class);
+                if (userTypeAdapter instanceof UserTypeAdapter &&
+                    metadataTypeAdapter instanceof MetadataTypeAdapter) {
+                    return (TypeAdapter<T>) new CommentTypeAdapter((UserTypeAdapter) userTypeAdapter,
+                                                                   (MetadataTypeAdapter) metadataTypeAdapter);
+                }
+            }
+
+            // by returning null, Gson will never check this factory if it can handle this TypeToken again [KZ] 6/15/16
+            return null;
+        }
```

Comment:
I wasn't able to think of a good way since each class that needs a TypeAdapterFactory needs it because it's TypeAdapter needs other typeadapters passed in.